soc/espressif/esp32/soc.c

```diff
@@ -18,6 +18,7 @@
 
 extern void z_prep_c(void);
 extern void esp_reset_reason_init(void);
+extern FUNC_NORETURN void z_cstart(void);
```

Comment:
With how things are currently structured, declaring z_cstart() here is not necessary as it is not called within this file. However, z_prep_c() is called, which in turn calls z_cstart(). This also applies to esp32s2 and esp32s3.